### PR TITLE
Use -miphonesimulator-version-min when building App.framework for simulator

### LINKED ISSUE #73384: Flutter cannot build against ARM simulators only in xcode
I'm building an app with dev and prod flavors. Android works fine but ios is facing a bit of trouble.

What I did as follows:
1. created two dev and prod schemes and two targets respectively.
2. add debug and release config files for different flavors
3. add plugin 'flutter_pdfview' in flutter and run pub get.

Things work fine when run through terminals
`flutter run --flavor prod` or `flutter run --flavor prod`

But when I build it through xcode, I'm getting this error
```
Ld /Users/pengfeihuang/Library/Developer/Xcode/DerivedData/Runner-ducgzqqypwgexiekvbigdedzqhbw/Build/Intermediates.noindex/Pods.build/Debug-dev-iphonesimulator/flutter_pdfview.build/Objects-normal/arm64/Binary/flutter_pdfview normal arm64 (in target 'flutter_pdfview' from project 'Pods')
    cd /Users/pengfeihuang/Developer/Flutter/projects/flavors/ios/Pods
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -target arm64-apple-ios9.0-simulator -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.0.sdk -L/Users/pengfeihuang/Library/Developer/Xcode/DerivedData/Runner-ducgzqqypwgexiekvbigdedzqhbw/Build/Products/Debug-dev-iphonesimulator/flutter_pdfview -F/Users/pengfeihuang/Library/Developer/Xcode/DerivedData/Runner-ducgzqqypwgexiekvbigdedzqhbw/Build/Products/Debug-dev-iphonesimulator/flutter_pdfview -F/Users/pengfeihuang/Developer/Flutter/projects/flavors/ios/Pods/../Flutter -filelist /Users/pengfeihuang/Library/Developer/Xcode/DerivedData/Runner-ducgzqqypwgexiekvbigdedzqhbw/Build/Intermediates.noindex/Pods.build/Debug-dev-iphonesimulator/flutter_pdfview.build/Objects-normal/arm64/flutter_pdfview.LinkFileList -install_name @rpath/flutter_pdfview.framework/flutter_pdfview -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -rpath -Xlinker @loader_path/Frameworks -dead_strip -Xlinker -object_path_lto -Xlinker /Users/pengfeihuang/Library/Developer/Xcode/DerivedData/Runner-ducgzqqypwgexiekvbigdedzqhbw/Build/Intermediates.noindex/Pods.build/Debug-dev-iphonesimulator/flutter_pdfview.build/Objects-normal/arm64/flutter_pdfview_lto.o -Xlinker -objc_abi_version -Xlinker 2 -fobjc-arc -fobjc-link-runtime -framework Flutter -framework Foundation -compatibility_version 1 -current_version 1 -Xlinker -dependency_info -Xlinker /Users/pengfeihuang/Library/Developer/Xcode/DerivedData/Runner-ducgzqqypwgexiekvbigdedzqhbw/Build/Intermediates.noindex/Pods.build/Debug-dev-iphonesimulator/flutter_pdfview.build/Objects-normal/arm64/flutter_pdfview_dependency_info.dat -o /Users/pengfeihuang/Library/Developer/Xcode/DerivedData/Runner-ducgzqqypwgexiekvbigdedzqhbw/Build/Intermediates.noindex/Pods.build/Debug-dev-iphonesimulator/flutter_pdfview.build/Objects-normal/arm64/Binary/flutter_pdfview

ld: building for iOS Simulator, but linking in dylib built for iOS, file '/Users/pengfeihuang/Developer/Flutter/projects/flavors/ios/Flutter/Flutter.framework/Flutter' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I saw similar issue [here](https://github.com/flutter/flutter/issues/64502), but I don't quite understand why it works with command line but not on xcode.

I have created a sample app [here](https://github.com/PandaGeek1024/flavors).

```
[✓] Flutter (Channel stable, 1.22.5, on Mac OS X 10.15.7 19H2 darwin-x64, locale en-NZ)
    • Flutter version 1.22.5 at /Users/pengfeihuang/fvm/versions/stable
    • Framework revision 7891006299 (4 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/pengfeihuang/Library/Android/sdk
    • Platform android-30, build-tools 29.0.2
    • ANDROID_HOME = /Users/pengfeihuang/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0, Build version 12A7209
    • CocoaPods version 1.10.0

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin installed
    • Dart plugin version 201.9317
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.0

[✓] Connected device (1 available)
    • iPhone SE (2nd generation) (mobile) • 01768AA3-066A-483C-BAED-91EDC3E50665 • ios • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)

• No issues found!

```

### PR DESCRIPTION
**Pull request by jmagman**
I ran into problems with this while trying to get ARM simulators working.

1. Compile debug App.framework stub with `-miphonesimulator-version-min` on simulator instead of `-miphoneos-version-min`.
2. Refactor `environmentTypeFromSdkroot` to take a string instead of coercing into a directory and back since a string is what's actually passed into `assemble`.
3. `cc` with `-miphoneos-version-min` for profile and release App.frameworks.  Slightly simplify flag logic by moving `-isysroot` flag into `commonBuildOptions` list since it's being passed into both `cc` and `clang`.

Fixes https://github.com/flutter/flutter/issues/73384
Related to https://github.com/flutter/flutter/issues/46125 and https://github.com/flutter/flutter/pull/46130

### CODE REVIEW

File: packages/flutter_tools/lib/src/base/build.dart [hunk 1]
```diff
@@ -260,14 +260,14 @@ class AOTSnapshotter {
...
-      isysrootArgs = <String>['-isysroot', sdkRoot];
+      commonBuildOptions.addAll(<String>[
+        '-isysroot', sdkRoot,
+      ]);
     }
 
     final RunResult compileResult = await _xcode.cc(<String>[
-      '-arch', targetArch,
-      if (isysrootArgs != null) ...isysrootArgs,
+      ...commonBuildOptions,
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/base/build.dart:270]: `commonBuildOptions` is `-arch`, `-miphoneos-version-min`, and `-isysroot`.

File: packages/flutter_tools/test/general.shard/base/build_test.dart [hunk 2]
```diff
@@ -29,26 +29,8 @@ const FakeCommand kARMCheckCommand = FakeCommand(
...
-  '-install_name',
-  '@rpath/App.framework/App',
   '-isysroot',
   'path/to/sdk',
-  '-o',
-  'build/foo/App.framework/App',
-  'build/foo/snapshot_assembly.o',
-];
-
-const List<String> kBitcodeClang = <String>[
```
Comment (jmagman) [inline packages/flutter_tools/test/general.shard/base/build_test.dart:50]: Moved these flags into the one test where they were used.

File: packages/flutter_tools/lib/src/base/build.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/base/build.dart:270]: nit: hoist this up to where commonBuildOptions is declared?

```
    final List<String> commonBuildOptions = <String>[
      '-arch', targetArch,
      if (isIOS)
        // When the minimum version is updated, remember to update
        // template MinimumOSVersion.
        // https://github.com/flutter/flutter/pull/62902
        '-miphoneos-version-min=8.0',
        if (sdkRoot != null) '-isysroot=$sdkroot',
    ];
```

**Review by jonahwilliams — APPROVED** [review]
LGTM with nit

File: packages/flutter_tools/lib/src/base/build.dart (hunk 1 above)
Comment (jmagman) [reply packages/flutter_tools/lib/src/base/build.dart:270]: Ha oh yeah that's better...

**Review by dnfield — APPROVED** [review]
LGTM

**Pull request merged**